src/plugins/dam/components/video-asset-picker-modal/index.jsx

```diff
@@ -0,0 +1,92 @@
+import NodeRef from '@gdbots/schemas/gdbots/ncr/NodeRef';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Uploader from '@triniti/cms/plugins/dam/components/uploader';
+import VideoAssetSearch from '@triniti/cms/plugins/dam/components/video-asset-search';
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+} from '@triniti/admin-ui-plugin/components';
+
+class VideoAssetPickerModal extends React.Component {
+  static propTypes = {
+    isOpen: PropTypes.bool,
+    onSelect: PropTypes.func.isRequired,
+    onToggleModal: PropTypes.func.isRequired,
+    onToggleUploader: PropTypes.func.isRequired,
+    refreshSearchFlag: PropTypes.number.isRequired, // on change, triggers new search
+    selected: PropTypes.arrayOf(PropTypes.instanceOf(NodeRef)),
+  };
+
+  static defaultProps = {
+    isOpen: false,
+    selected: [],
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isUploaderOpen: false,
+    };
+    this.handleToggleUploader = this.handleToggleUploader.bind(this);
+  }
+
+  handleToggleUploader(...args) {
+    const { onToggleUploader } = this.props;
+    this.setState(({ isUploaderOpen }) => ({
+      isUploaderOpen: !isUploaderOpen,
+    }), () => onToggleUploader(...args));
+  }
+
+  render() {
+    const {
+      isOpen,
+      onSelect: handleSelect,
+      onToggleModal: handleToggleModal,
+      onToggleUploader: handleToggleUploader,
+      selected,
+      refreshSearchFlag,
+    } = this.props;
+    const { isUploaderOpen } = this.state;
+
+    return (
+      <div>
+        <Modal centered size="xxl" isOpen={isOpen} toggle={handleToggleModal}>
+          <ModalHeader toggle={handleToggleModal}>
+            <span className="nowrap">Select A Video Asset</span>
+          </ModalHeader>
+          <ModalBody className="p-0">
+            <VideoAssetSearch
+              onSelect={handleSelect}
+              selected={selected}
+              refreshSearchFlag={refreshSearchFlag}
+              onToggleUploader={handleToggleUploader}
+            />
+          </ModalBody>
+          <ModalFooter>
+            <Button className="mr-auto" color="primary" onClick={this.handleToggleUploader}>
+              Upload
+            </Button>
+            <Button onClick={handleToggleModal} disabled={!selected.length}>
+              Select
+            </Button>
+          </ModalFooter>
+        </Modal>
+        {isUploaderOpen && (
+        <Uploader
+          allowedMimeTypes={['video/x-flv', 'video/mp4', 'video/webm']}
```

Comment:
@efjacobson what about mxf?

Reply:
@gdbrown we can add it but its not supported by jwplayer, see `sourceformat ` at https://developer.jwplayer.com/jwplayer/reference#post_videos-create

since this isnt used by anything else but jwplayer it seemed ok for now. can change to a prop later, or now if you like

Reply:
@efjacobson hmmm, i would imagine the mezz ref is always going to be mxf though but we'd resolve it when we send to jwplayer as a mp4 ref that they can use.  so, we should allow mxf here

Reply:
@gdbrown done